flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosApplicationMasterRunner.java

```diff
@@ -601,6 +601,54 @@ else if (recoveryMode == HighAvailabilityMode.ZOOKEEPER) {
 
 		info.setCommand(cmd);
 
+		// Set base container for task manager if specified in configs.
+		String taskManagerContainerName = flinkConfig.getString(
+			ConfigConstants.MESOS_RESOURCEMANAGER_TASKS_CONTAINER_IMAGE_NAME, "");
+
+		if (taskManagerContainerName.length() > 0) {
+			String taskManagerContainerType = flinkConfig.getString(
+				ConfigConstants.MESOS_RESOURCEMANAGER_TASKS_CONTAINER_TYPE,
+				ConfigConstants.DEFAULT_MESOS_RESOURCEMANAGER_TASKS_CONTAINER_IMAGE_TYPE);
+
+			Protos.ContainerInfo.Builder containerInfo;
+
+			switch (taskManagerContainerType) {
+				case ConfigConstants.MESOS_RESOURCEMANAGER_TASKS_CONTAINER_TYPE_MESOS:
+					containerInfo = Protos.ContainerInfo.newBuilder()
+						.setType(Protos.ContainerInfo.Type.MESOS)
+						.setMesos(Protos.ContainerInfo.MesosInfo.newBuilder()
+							.setImage(Protos.Image.newBuilder()
+								.setType(Protos.Image.Type.DOCKER)
+								.setDocker(Protos.Image.Docker.newBuilder()
+									.setName(taskManagerContainerName))));
+					break;
+				case ConfigConstants.MESOS_RESOURCEMANAGER_TASKS_CONTAINER_TYPE_DOCKER:
+					containerInfo = Protos.ContainerInfo.newBuilder()
+						.setType(Protos.ContainerInfo.Type.DOCKER)
+						.setDocker(Protos.ContainerInfo.DockerInfo.newBuilder()
+							.setNetwork(Protos.ContainerInfo.DockerInfo.Network.HOST)
+							.setForcePullImage(true)
```

Comment:
Then you can specify the image name with your tag: `myimage:devel`
But we can make a setting for it which defaults to true (I think force-pulling should be on by default, as it has no side-effects).